app/controllers/trips_controller.rb

```diff
@@ -0,0 +1,94 @@
+class TripsController < ApplicationController
+
+  def index
+    @trips = Trip.all
+  end
+
+  def show
+    @trip = Trip.find_by(id: params[:id])
+
+    if @trip.nil?
+      head :not_found
+      return
+    end
+  end
+
+  def create #check for accuracy
+    passenger = Passenger.find_by(id: params[:passenger_id])
+    if passenger.nil?
+      head :not_found
+      return
+    else
+      cost = Trip.trip_cost
+      driver = Driver.assign_driver
+      trip = Trip.new(driver_id: driver.id, passenger_id: passenger.id, date: Date.today, rating: nil, cost: cost)
+      if trip.save
+        redirect_to trip_path(trip.id)
+        return
+      else
+        redirect_to passenger_path(passenger.id)
```

Comment:
redirect?  Why not do this?

```suggestion
        render :new, status: :bad_request
```